src/Test/Utilities/Portable/TempFiles/TempRoot.cs

```diff
@@ -1,15 +1,15 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
+using System.Collections.Concurrent;
 using System.IO;
 using System.Runtime.CompilerServices;
 
 namespace Microsoft.CodeAnalysis.Test.Utilities
 {
     public sealed class TempRoot : IDisposable
     {
-        private readonly List<IDisposable> _temps = new List<IDisposable>();
+        private readonly ConcurrentBag<IDisposable> _temps = new ConcurrentBag<IDisposable>();
```

Comment:
I prefer the earlier version of this change. This type was not designed with thread safety in mind and I'm skeptical of making it more free threaded. In particular I'm always skeptical of types which are a) disposable and b) writable from multiple threads.

Reply:
Okay, then you and @jasonmalinowski need to talk. Both of you are giving conflicting feedback.

Reply:
My major objection comes down to Dispose. Mutation after Dispose should be an error. Enforcing that contract though adds a bit more state and complexity to the TempRoot object. Enough that I question if it's worth it to support the single case where mutate TempRoot from multiple threads vs. just adding the lock in that one case.

Reply:
@jaredpar If you can prove that was the only test in our entire codebase using this in a multi-threaded way, I'll be OK with the original fix. Keep in mind this tool is used beyond just the compiler tests, so don't restrict your statements to that. Otherwise, let's just fix it for good *before* we create another flaky test down the road.

I don't want perfect to be the enemy of "leaving traps for the next person" here.

Reply:
Okay, but what about the IDE tests Jason is talking about? Discussions about IDisposable aren't really relevant (it's simple enough to add tracking vs. finding/fixing all the IDE tests, and/or validating the IDE tests aren't broken)